File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    maxprofit := 0
    left := 0
    for right := 1; right < len(prices); right++ {
        if prices[left] > prices[right] {
            left = right
        } else {
            maxprofit = max(maxprofit, prices[right]-prices[left])
        }
    }
    return maxprofit
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0;
        int left = 0;
        for (int right = 1; right < prices.size(); ++right) {
            if (prices[left] > prices[right]) {
                left = right;
            } else {
                maxprofit = max(maxprofit, prices[right] - prices[left]);
            }
        }
        return maxprofit;
    }
};
 